openedx/core/djangoapps/schedules/signals.py

```diff
@@ -0,0 +1,74 @@
+import datetime
+import logging
+
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from django.utils import timezone
+
+from course_modes.models import CourseMode
+from courseware.models import DynamicUpgradeDeadlineConfiguration, CourseDynamicUpgradeDeadlineConfiguration
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from student.models import CourseEnrollment
+from .models import Schedule
+
+log = logging.getLogger(__name__)
+
+
+def get_upgrade_deadline(enrollment):
```

Comment:
This schedules API seems like it should go in another python module (maybe api.py?).